scripts/Tools/xmlquery

```diff
@@ -163,12 +189,40 @@ def _main_func():
     # Initialize command line parser and get command line options
     variables, subgroup, caseroot, listall,  fileonly, \
         value, no_resolve, raw, description, group, full, dtype, \
-        valid_values = parse_command_line(sys.argv)
+        valid_values, partial_match = parse_command_line(sys.argv)
 
     # Initialize case ; read in all xml files from caseroot
     with Case(caseroot) as case:
-        if listall:
-            variables = sorted(case.get_record_fields(None, "varid"))
+        if listall or partial_match:
+            all_variables = sorted(case.get_record_fields(None, "varid"))
+            if partial_match:
+                all_matching_vars = []
+                for variable in variables:
+                    for all_variable in all_variables:
+                        regex = re.compile(variable)
+                        if regex.match(all_variable):
```

Comment:
Why regex.match instead of regex.search? That way, I can do something like:
`./xmlquery -p _ATM`
and get all the ATM variables.